paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java

```diff
@@ -195,7 +195,16 @@ private static List<ManifestFileMeta> allManifests(FileStoreTable dataTable) {
         SnapshotManager snapshotManager = dataTable.snapshotManager();
         Long snapshotId = coreOptions.scanSnapshotId();
         Snapshot snapshot = null;
-        if (snapshotId != null && snapshotManager.snapshotExists(snapshotId)) {
+        if (snapshotId != null) {
+            // reminder user with snapshot id range
+            if (!snapshotManager.snapshotExists(snapshotId)) {
+                Long earliestSnapshotId = snapshotManager.earliestSnapshotId();
+                Long latestSnapshotId = snapshotManager.latestSnapshotId();
+                throw new RuntimeException(
+                        String.format(
+                                "scan.snapshot-id is not exist, you can set it in range from %s to %s",
+                                earliestSnapshotId, latestSnapshotId));
```

Comment:
better : Specified scan.snapshot-id [%s] is not exist, .....

Reply:
Thanks，your advice is nice. @LinMingQiang